packages/state-transition/src/block/processBlsToExecutionChange.ts

```diff
@@ -0,0 +1,43 @@
+import {capella, ssz} from "@lodestar/types";
+import {BLS_WITHDRAWAL_PREFIX, ETH1_ADDRESS_WITHDRAWAL_PREFIX, WITHDRAWAL_PREFIX_BYTES} from "@lodestar/params";
+import {toHexString, byteArrayEquals} from "@chainsafe/ssz";
+import {digest} from "@chainsafe/as-sha256";
+
+import {CachedBeaconStateCapella} from "../types.js";
+
+export function processBlsToExecutionChange(
+  state: CachedBeaconStateCapella,
+  signedBlsToExecutionChange: capella.SignedBLSToExecutionChange
+): void {
+  const addressChange = signedBlsToExecutionChange.message;
+  if (addressChange.validatorIndex >= state.validators.length) {
+    throw Error(`Invalid validatorIndex expected<${state.validators.length} actual=${addressChange.validatorIndex}`);
+  }
+  const validator = state.validators.get(addressChange.validatorIndex);
+  // We need to work on the slice otherwise ssz view seems to be getting messed up
+  // once we modify this using set for putting new credentials
+  const withdrawalCredentials = validator.withdrawalCredentials.slice();
+  const credentialPrefix = withdrawalCredentials.slice(0, WITHDRAWAL_PREFIX_BYTES);
+  if (!byteArrayEquals(credentialPrefix, BLS_WITHDRAWAL_PREFIX)) {
```

Comment:
BLS_WITHDRAWAL_PREFIX is a single byte so you only need to do
```suggestion
  if (withdrawalCredentials[0] !== BLS_WITHDRAWAL_PREFIX) {
```
And declare BLS_WITHDRAWAL_PREFIX as a number instead of Uint8Array